internal/build/iana_rdap.go

```diff
@@ -0,0 +1,49 @@
+package build
+
+import (
+	"encoding/json"
+)
+
+const (
+	ianaRdapURL = "https://data.iana.org/rdap/dns.json"
+)
+
+// FetchRDAPFromIANA retrieves the map of zones to RDAP service endpoints from IANA.
+func FetchRDAPFromIANA(zones map[string]*Zone) error {
+	res, err := Fetch(ianaRdapURL)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	x := struct {
+		Services [][][]string // I'm screaming at this schema
+	}{}
+	d := json.NewDecoder(res.Body)
+	err = d.Decode(&x)
+	if err != nil {
+		return err
+	}
+	for _, svc := range x.Services {
+		tlds := svc[0]
+
+		for _, tld := range tlds {
+			z, ok := zones[tld]
+			if !ok {
+				continue
```

Comment:
Thoughts on what we should do if the zone/TLD is missing in the map?

Reply:
Log a warning to stderr and continue. There should be examples of this elsewhere. e.g. if we identify a domain that’s *not* in ZoneDB (rare, but possible), we can fix it.